plugins/inputs/sflow/decoder/funcs.go

```diff
@@ -0,0 +1,246 @@
+package decoder
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf/metric"
+)
+
+// U32 answers a directive for 32bit Unsigned Integers
+func U32() ValueDirective {
+	return &valueDirective{value: new(uint32), resetFn: func(in interface{}) {
+		ui32Ptr, ok := in.(*uint32)
+		if !ok {
+			// Can't be tested
+			panic("must be an *uint32")
```

Comment:
this panic isn't very useful. You would have gotten a better error message if you just did `ui32Ptr := in.(*uint32)` and accepted the default panic.